src/db.rs

```diff
@@ -266,6 +267,51 @@ impl DbInner {
         }
     }
 
+    #[allow(clippy::panic)]
+    pub async fn write_batch_with_options(&self, batch: &WriteBatch, options: &WriteOptions) {
+        self.maybe_apply_backpressure().await;
+
+        let current_table = if self.wal_enabled() {
+            let mut guard = self.state.write();
+            let current_wal = guard.wal();
+            for op in &batch.ops {
+                match op {
+                    WriteOp::Put(key, value) => {
+                        current_wal.put(key, value);
+                    }
+                    WriteOp::Delete(key) => {
+                        current_wal.delete(key);
+                    }
+                }
+            }
+            current_wal.table().clone()
+        } else {
+            if cfg!(not(feature = "wal_disable")) {
+                panic!("wal_disabled feature must be enabled");
+            }
+            let mut guard = self.state.write();
+            let current_memtable = guard.memtable();
+            for op in &batch.ops {
+                match op {
+                    WriteOp::Put(key, value) => {
+                        current_memtable.put(key, value);
+                    }
+                    WriteOp::Delete(key) => {
+                        current_memtable.delete(key);
+                    }
+                }
+            }
+            let table = current_memtable.table().clone();
+            let last_wal_id = guard.last_written_wal_id();
+            self.maybe_freeze_memtable(&mut guard, last_wal_id);
```

Comment:
What if the batch is too large? We may cause the Memtable to exceed the predetermined size, cockroach has related discussions.
[Cockroach Large Batches](https://github.com/cockroachdb/pebble/blob/master/docs/rocksdb.md#large-batches)

Reply:
I personally recommend introducing relevant functions before writing the batch to Memtable to ensure that there is enough space in Memtable.

Reply:
We currently have no SST size limit for WAL writes. They're just flushed every N ms. We did discuss adding such a config in #30, but opted against it at the time. In cases where `wal_disabled` is set, I'm thinking we will simply allow the batch to overrun the `l0_sst_size_bytes` setting.

Reply:
It sounds like, loosely, Pebble does this as well.

> A batch that is too large to fit in the memtable causes the current mutable memtable to be marked as immutable and the batch is wrapped in a flushableBatch structure and added to the list of immutable memtables. Because the flushableBatch is readable as another layer in the LSM, the batch commit can return as soon as the flushableBatch has been added to the immutable memtable list.

It sounds like they take an entire batch and treat it as a memtable if it's too large to fit into the current memtable.

Reply:
Resolving this. LMK if you have any concerns about unlimited batch sizes.